addons/mail/static/src/owl/store/messaging_actions.js

```diff
@@ -778,6 +778,74 @@ const actions = {
         }
         thread.message_unread_counter = 0;
     },
+    /**
+     * @param {Object} param0
+     * @param {function} param0.dispatch
+     * @param {Object} param0.state
+     * @param {string[]} messageLocalIds
+     */
+    async moderationAccept({ dispatch, state }, messageLocalIds) {
+        return dispatch('_moderateMessages', messageLocalIds, 'accept');
+    },
+    /**
+     * @param {Object} param0
+     * @param {function} param0.dispatch
+     * @param {Object} param0.state
+     * @param {string[]} messageLocalIds
+     */
+    async moderationAllow({ dispatch, state }, messageLocalIds) {
+        return dispatch('_moderateMessages', messageLocalIds, 'allow');
+    },
+    /**
+     * @param {Object} param0
+     * @param {function} param0.dispatch
+     * @param {Object} param0.state
+     * @param {string[]} messageLocalIds
+     */
+    async moderationBan({ dispatch, state }, messageLocalIds) {
+        return dispatch('_moderateMessages', messageLocalIds, 'ban');
+    },
+    /**
+     * @param {Object} param0
+     * @param {function} param0.dispatch
+     * @param {Object} param0.state
+     * @param {string[]} messageLocalIds
+     */
+    async moderationDiscard({ dispatch, state }, messageLocalIds) {
```

Comment:
- unused `param0.state`
- I would merge all these actions into a single one, `moderateMessages`, with data to tell which `type` of moderation action is applied on these messages

Reply:
ok

Reply:
(basically make _moderateMessages public and use it)